cedar-example-use-cases/quip/quip.cedarschema

```diff
@@ -0,0 +1,78 @@
+// Entities
+// All enums are modeled as just String here
+
+entity User {
+    id: String,
+    name: String,
+    disabled: Bool,
+    shared_folder_ids?: Set<Folder>, // folders that are explicitly shared with the user
```

Comment:
Curious:  why is this field (and other Set-typed fields below) optional?  If there are no folders shared with  a user, this can be modeled as a required Set attribute, where the Set is empty.

Reply:
Quick conjectured answer (but curious about the correct answer): Maybe since it is possible to test that `shared_folder_ids` is present, but not (at the moment) possible to set whether a set is empty?

Reply:
Ah, good point; that makes sense.

Reply:
Thats a good point! I cant remember if I had tried the empty set but had remembered seeing the `has` keyword for policies with optional fields so went that route. Im still fairly novice with cedar syntax, but like Mike said I'm unsure if theres a way to test whether a set is empty.